packages/ef-runtime-client/src/EFRuntime/EFRuntime.ts

```diff
@@ -0,0 +1,87 @@
+import { IComponentRegistry } from "../ComponentRegistry";
+import { ModuleLoader } from "../ModuleLoader";
+import { StylingHandler } from "../StylingHandler";
+
+export interface IRuntimeDependencies {
+  componentRegistry: IComponentRegistry;
+  moduleLoader: ModuleLoader;
+  stylingHandler: StylingHandler;
+}
+
+export class EFRuntime {
+  private registry: IComponentRegistry;
+  private moduleLoader: ModuleLoader;
+  private stylingHandler: StylingHandler;
+
+  constructor({
+    componentRegistry,
+    moduleLoader,
+    stylingHandler,
+  }: IRuntimeDependencies) {
+    this.registry = componentRegistry;
+    this.moduleLoader = moduleLoader;
+    this.stylingHandler = stylingHandler;
+  }
+
+  private validateOptions(options: {
+    systemCode?: string;
+    overrides?: { [propName: string]: string };
+  }): void {
+    if (!options.systemCode) {
+      throw new Error("Must provide a systemCode option");
+    }
+  }
+
+  async init(
+    options: {
+      systemCode?: string;
+      overrides?: { [propName: string]: string };
+    } = {}
+  ): Promise<void> {
+    this.validateOptions(options);
+
+    await Promise.all([
+      this.moduleLoader.init(),
+      this.registry.fetch(options.systemCode as string),
+    ]);
+
+    if (options.overrides) {
+      this.registry.applyOverrides(options.overrides);
+    }
+
+    this.loadAll();
+  }
+
+  loadAll(): void {
+    for (const component of this.registry.getComponentKeys()) {
```

Comment:
I think you can just use `for (const component in this.registry)` to loop over the keys of an Object.